activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +88,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| format(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def format(arg)
+          case arg
+          when Hash
+            arg.transform_values { |value| format(value) }
+          when Array
+            arg.map { |value| format(value) }
+          when GlobalID::Identification
+            arg.to_global_id rescue arg
```

Comment:
An unpersisted model doesn't necessarily lack an `id`. I can arbitrarily assign one to a model, which causes the `to_global_id` function to work.

Anyway, here's the correction you suggested.